Lib/multiprocessing/shared_memory.py

```diff
@@ -0,0 +1,573 @@
+"Provides shared memory for direct access across processes."
+
+
+__all__ = [ 'SharedMemory', 'PosixSharedMemory', 'WindowsNamedSharedMemory',
+            'ShareableList', 'shareable_wrap',
+            'SharedMemoryServer', 'SharedMemoryManager', 'SharedMemoryTracker' ]
+
+
+from functools import reduce
+import mmap
+from .managers import DictProxy, SyncManager, Server
+from . import util
+import os
+import random
+import struct
+import sys
+try:
+    from _posixshmem import _PosixSharedMemory, Error, ExistentialError, O_CREX
+except ImportError as ie:
+    if os.name != "nt":
+        # On Windows, posixshmem is not required to be available.
+        raise ie
+    else:
+        _PosixSharedMemory = object
+        class ExistentialError(BaseException): pass
+        class Error(BaseException): pass
+        O_CREX = -1
+
+
+class WindowsNamedSharedMemory:
+
+    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
+        if name is None:
+            name = f'wnsm_{os.getpid()}_{random.randrange(100000)}'
+
+        self._mmap = mmap.mmap(-1, size, tagname=name)
+        self.buf = memoryview(self._mmap)
+        self.name = name
+        self.size = size
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.name!r}, size={self.size})'
+
+    def close(self):
+        self.buf.release()
+        self._mmap.close()
+
+    def unlink(self):
+        """Windows ensures that destruction of the last reference to this
+        named shared memory block will result in the release of this memory."""
+        pass
+
+
+class PosixSharedMemory(_PosixSharedMemory):
+
+    def __init__(self, name, flags=None, mode=None, size=None, read_only=False):
+        if name and (flags is None):
+            _PosixSharedMemory.__init__(self, name)
+        else:
+            if name is None:
+                name = f'psm_{os.getpid()}_{random.randrange(100000)}'
+            _PosixSharedMemory.__init__(self, name, flags=O_CREX, size=size)
+
+        self._mmap = mmap.mmap(self.fd, self.size)
+        self.buf = memoryview(self._mmap)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}({self.name!r}, size={self.size})'
+
+    def close(self):
+        self.buf.release()
+        self._mmap.close()
+        self.close_fd()
+
+
+class SharedMemory:
```

Comment:
I personally don't like this pattern. This is basically a function call, and could also just be an alias. Something like:

```
if os.name == 'nt':
    SharedMemory = WindowsSharedMemory
else:
    SharedMemory = PosixSharedMemory
```

Reply:
Removed in GH-11816.